README.md

```diff
@@ -396,40 +516,210 @@ Make sure that you have specified `authenticationEndpoint` during SDK initializa
+        .setMaxRetries(5)
+        .build(),
+    preprocessor = ImagePreprocess.Builder()
+        .limit(512, 512)
+        .rotate(90f)
+        .build(),
+    imageKitCallback = object: ImageKitCallback {
         override fun onSuccess(uploadResponse: UploadResponse) {
-                // Handle Success Response
+            // Handle Success Response
         }
         override fun onError(uploadError: UploadError) {
-                // Handle Upload Error
+            // Handle Upload Error
         }
     }
 )
 ```
 
+### UploadPolicy
+The `UploadPolicy` class represents a set of conditions that need to be met for an upload request to be executed.
+
+`UploadPolicy.Builder` class is responsible for building the UploadPolicy instances. This class provides following methods to access and modify the policy parameters:
+
+| Parameter                                          | Type                 | Description                                                                                                                                                                                                                      |
+|:---------------------------------------------------|:---------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| requireNetworkType(type: NetworkType )             | UploadPolicy.Builder | Specifies the network type required for the upload request. Possible values are `UploadPolicy.NetworkPolicy.ANY` and `UploadPolicy.NetworkPolicy.UNMETERED`. Defaults to `NetworkPolicy.ANY`.                                    |
+| requiresBatteryCharging(requiresCharging: Boolean) | UploadPolicy.Builder | Sets whether the device needs to be connected to a charger for the upload request. Defaults to `false`.                                                                                                                          |
+| requiresDeviceIdle(requiresIdle: Boolean)          | UploadPolicy.Builder | Sets whether the device needs to be idle for the upload request. Defaults to `false`.                                                                                                                                            |
+| setMaxRetries(count: Int)                          | UploadPolicy.Builder | Sets the maximum number of retries for the upload request. Negative value will throw an `IllegalArgumentException`. Defaults to 5.                                                                                               |
+| setRetryBackoff(interval, BackoffPolicy policy)    | UploadPolicy.Builder | Sets the backoff interval in milliseconds and policy (`UploadPolicy.BackoffPolicy.LINEAR` or `UploadPolicy.BackoffPolicy.EXPONENTIAL`) for retry attempts. Defaults to interval of 10000ms and policy of `BackoffPolicy.LINEAR`. |
+
+Example code
```

Comment:
In addition to code, explain what this does in a paragraph.

Reply:
Done